src/error/error_kinds.rs

```diff
@@ -0,0 +1,43 @@
+use std::io;
+
+/// Errors that could occur with constructing parsing packet contents
+#[derive(Fail, Debug, PartialEq, Eq, Clone)]
+pub enum PacketErrorKind {
+    #[fail(display = "Could not process the packet, because the packet size was bigger than the max allowed size.")]
```

Comment:
I would remove the `could not process the packet, because` because its extra and its repeated. I would just have it say "The packet size was bigger than the max allowed size.`